src/mds/FSMap.cc

```diff
@@ -1006,6 +1006,12 @@ void FSMap::erase(mds_gid_t who, epoch_t blocklist_epoch)
         // the rank ever existed so that next time it's handed out
         // to a gid it'll go back into CREATING.
         fs.mds_map.in.erase(info.rank);
+      } else if (info.state == MDSMap::STATE_STARTING) {
```

Comment:
This seems to be an ideal place to introduce kill points. @batrick WDYT?

Reply:
I think you mean in MDSRank during `up:starting`? Probably a good idea, yes. The MDSThrasher could set various killpoints during workload testing too.